huaweicloud/data_source_huaweicloud_iec_bandwidths.go

```diff
@@ -0,0 +1,126 @@
+package huaweicloud
+
+import (
+	"context"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/logp"
+
+	"github.com/chnsz/golangsdk/openstack/iec/v1/bandwidths"
+	iec_common "github.com/chnsz/golangsdk/openstack/iec/v1/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+)
+
+func dataSourceIECBandWidths() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceIECBandWidthsRead,
+
+		Schema: map[string]*schema.Schema{
+			"site_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"site_info": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"bandwidths": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"size": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"share_type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"charge_mode": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"line": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceIECBandWidthsRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	iecClient, err := config.IECV1Client(GetRegion(d, config))
+	if err != nil {
+		return fmtp.DiagErrorf("Error creating Huaweicloud IEC client: %s", err)
+	}
+
+	listOpts := bandwidths.ListOpts{
+		SiteID: d.Get("site_id").(string),
+	}
+
+	allBWs, err := bandwidths.List(iecClient, listOpts).Extract()
+	if err != nil {
+		return fmtp.DiagErrorf("Unable to extract iec bandwidths: %s", err)
+	}
+
+	total := len(allBWs.BandWidth)
+	if total < 1 {
```

Comment:
better to not raising error for listing data source.

Reply:
fixed

```
$ make testacc TEST='./huaweicloud' TESTARGS='-run TestAccIECBandWidthsDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run TestAccIECBandWidthsDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccIECBandWidthsDataSource_basic
=== PAUSE TestAccIECBandWidthsDataSource_basic
=== CONT  TestAccIECBandWidthsDataSource_basic
--- PASS: TestAccIECBandWidthsDataSource_basic (119.02s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       119.765s
```